### PR TITLE
add version and pydebug patch

### PR DESCRIPTION
**Pull request by joequant**

### CODE REVIEW

**Review by vvolkl — CHANGES_REQUESTED** [review]
Thanks @joequant, the new version has already been added in https://github.com/spack/spack/pull/29744, please rebase on that.

Also please prepend the PR title with "py-awkward: " for a more readable commit history.

Can you add a comment motivating the patch, with a link to an upstream issue/commit if that exists?

File: var/spack/repos/builtin/packages/py-awkward/pydebug.patch [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+@@ -177,10 +177,11 @@ class Install(setuptools.command.install.install):
+         outerdir = os.path.join(
+             os.path.join(
+                 "build",
+-                "lib.{}-{}.{}".format(
++                "lib.{}-{}.{}{}".format(
+                     get_platform(),
+                     sys.version_info[0],
+                     sys.version_info[1],
++                    'd' in sys.abiflags ? "-pydebug" : ""
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-awkward/pydebug.patch]: Is this even valid Python?

Comment (joequant) [reply var/spack/repos/builtin/packages/py-awkward/pydebug.patch]: thinking c++ let me put in a fix

**Comment by joequant** [comment]
yuck..  Let me force a clean patch.

**Comment by alalazo** [comment]
Still waiting for:

> Also please prepend the PR title with "py-awkward: " for a more readable commit history.
>
> Can you add a comment motivating the patch, with a link to an upstream issue/commit if that exists?

Closing in the meanwhile, but feel free to reopen if you want to continue working on this.

**Pull request closed without merging**